### PR TITLE
Ensure you can't request to join a team you've already asked to join!

### PR DESCRIPTION
**Pull request by idrysdale**
Check the database and exit during `team join` if there are any existing requests to join the same team.

(Requires https://github.com/fluidkeys/fluidkeys/pull/397)

### CODE REVIEW

File: fk/teamjoin.go [hunk 1]
```diff
@@ -69,6 +75,27 @@ func teamJoin(teamUUID uuid.UUID) exitCode {
...
+	existingRequests, err := db.GetRequestsToJoinTeam(teamUUID)
+	if err != nil {
+		out.Print(ui.FormatFailure("Failed to check for existing requests", nil, err))
+		return 1
+	}
+	if len(existingRequests) > 0 {
+		out.Print(ui.FormatWarning(
+			"You've already requested to join "+existingRequests[0].TeamName,
+			[]string{
+				"You requested to join " + existingRequests[0].TeamName + " " +
```
Comment (paulfurley) [inline fk/teamjoin.go]: be great if you could use the helper function I've made for this, I'll put the commit as a separate pull request

Comment (idrysdale) [reply fk/teamjoin.go]: Wilco!

Comment (idrysdale) [reply fk/teamjoin.go]: Done!

File: fk/teamjoin.go [hunk 2]
```diff
@@ -31,6 +31,10 @@ import (
 )
 
 func teamJoin(teamUUID uuid.UUID) exitCode {
+	if exitCode := ensureNoExistingRequests(teamUUID); exitCode != 0 {
```
Comment (paulfurley) [inline fk/teamjoin.go]: Doesn't this want to be *after* we've worked out which key they're using? A request to join a key is unique on `(team UUID, fingerprint)`, not on `teamUUID`

Comment (idrysdale) [reply fk/teamjoin.go]: :) good thinking

**Pull request closed without merging**